pages/forms/TripForm/[id].js

```diff
@@ -0,0 +1,83 @@
+import CancelButton from "@/components/CancelButton";
+import CreateButton from "@/components/CreateButton";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+export default function TripDetailsForm() {
+  const { mutate } = useSWR("/api/trips");
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: trips, isLoading } = useSWR(
+    id ? `/api/trips/${id}` : null,
+    fetcher
+  );
+
+  if (!trips || isLoading) {
+    return <>is Loading...</>;
```

Comment:
To improve your HTML markup, use an `<h2> `or something similar


```suggestion
    return <h2> is loading...</h2>;
```